### PR TITLE
Fix ISO8601 bug

### PR DESCRIPTION
**Pull request by lucasmichot**
Fix ISO8601 BC

---

FIx https://github.com/briannesbitt/Carbon/issues/861

### CODE REVIEW

**Comment by dereuromark** [comment]
@lucasmichot I think your changes revert the BC break, so they "could" make sense after all.

**Comment by lucasmichot** [comment]
Yeah but is this really a BC - or more an expected behaviour as per your comment in #861 ? It's not that obvious
Ping @briannesbitt

**Comment by briannesbitt** [comment]
Ya I mentioned awhile ago that its a breaking change but also a bug fix to implement correct behaviour.  I guess the issue is people are relying on the incorrect behaviour so thats why its a break.

I had changed it to use the atom format awhile ago but I guess it never got released.  I guess save it for 2.0 at this point to mimic PHP behaviour.

**Comment by ameliaikeda** [comment]
It's probably worth noting that *technically* both formats are valid under ISO8601, but PHP's default handling of dates (`createFromFormat` being the biggest problem) doesn't allow parsing ISO8601 properly, as it's too rigid. Not everyone uses new/parse though, and interop for proper handling of iso8601 is terrible in general.

File: tests/Carbon/StringsTest.php [hunk 1]
```diff
@@ -81,7 +81,7 @@ public function testToLocalizedFormattedDateStringWhenUtf8IsNedded()
          * $cache = setlocale(LC_TIME, 0);
          * $d = Carbon::create(2016, 01, 06, 00, 00, 00);
          * setlocale(LC_TIME, 'spanish');
-         * $this->assertSame(utf8_encode('mi�rcoles 06 enero 2016'), $d->formatLocalized('%A %d %B %Y'));
+         * $this->assertSame(utf8_encode('mi�rcoles 06 enero 2016'), $d->formatLocalized('%A %d %B %Y'));
```
Comment (ameliaikeda) [inline tests/Carbon/StringsTest.php]: potential encoding error in your client here, maybe?

**Comment by lucasmichot** [comment]
@briannesbitt I think that even if PHP does not exactrly the ISO, it should be worth to fix this in `1.x`
Somehow it's globally accepted in the PHP community

**Comment by ameliaikeda** [comment]
@lucasmichot PHP isn't actually wrong, however: ISO8601 is a very large standard. Both `Y-m-d\TH:i:sO` and `Y-m-d\TH:i:sP` are correct under ISO8601, as is `Y-m-d`, `Y-m-d H:i:s` and `Y-m`. PHP just doesn't have very flexible handling by default.

**Comment by lucasmichot** [comment]
Thanks for making it clearer @ameliaikeda 
Neverheless we should maybe provide the default `\DateTime::ISO8601` format value

**Comment by jaylinski** [comment]
I wrote a little blog post on this issue: https://dev.karriere.at/a/so-you-want-to-use-iso-8601-dates-in-php-and-browsers

**Comment by jaylinski** [comment]
@Glavic You are right, this works in PHP and in browsers. I will update the article.

**Comment by kylekatarnls** [comment]
Won't fix as we do not wish to align on a PHP documented bug. If you need the legacy ISO8601 format, just use `->format(Carbon::ISO8601)`

**Pull request closed without merging**